File: s3-inf-ppo/laboratoria/lab07/php/src/QueueGenerator.php
<?php

namespace ParkingLotManager;

use ParkingLotManager\Entities\Car;
use ParkingLotManager\Entities\Bicycle;
use ParkingLotManager\Entities\Pedestrian;
use ParkingLotManager\Entities\TeacherCar;

final class QueueGenerator {

	const ANONYMOUS_PEDESTRIANS_COUNT = 50;
	const PEDESTRIANS_COUNT = 5;
	const CARS_COUNT = 30;
	const TEACHER_CARS_COUNT = 7;
	const BICYCLES_COUNT = 3;

	public static function generate(): array {
		$queue = [];

		for($i = 0; $i <= self::ANONYMOUS_PEDESTRIANS_COUNT; $i++) {
			$queue[] = new Pedestrian();
		}

		for($i = 0; $i <= self::PEDESTRIANS_COUNT; $i++) {
			$queue[] = new Pedestrian(self::getRandomName());
		}

		for($i = 0; $i <= self::CARS_COUNT; $i++) {
			$queue[] = new Car(self::getRandomPlateNumber());
		}

		for($i = 0; $i <= self::TEACHER_CARS_COUNT; $i++) {
			$queue[] = new TeacherCar(self::getRandomPlateNumber());
		}

		for($i = 0; $i <= self::BICYCLES_COUNT; $i++) {
			$queue[] = new Bicycle();
		}

		shuffle($queue);
		return $queue;
	}

	private static function getRandomPlateNumber(): string {
		return "DLX " . random_int(10000, 99999);
	}

	private static function getRandomName(): string {
		$names = ["John", "Jack", "James", "George", "Joe", "Jim"];
		return $names[array_rand($names)];
	}

}

File: s3-inf-ppo/laboratoria/lab07/java/ppo7/src/ParkingLotManager/QueueGenerator.java
package ParkingLotManager;

import ParkingLotManager.Entities.Bicycle;
import ParkingLotManager.Entities.Car;
import ParkingLotManager.Entities.Pedestrian;
import ParkingLotManager.Entities.TeacherCar;
import ParkingLotManager.Interfaces.EntityInterface;

import java.util.ArrayList;
import java.util.Collections;
import java.util.Random;

public class QueueGenerator {

    private static final int ANONYMOUS_PEDESTRIANS_COUNT = 50;
    private static final int PEDESTRIANS_COUNT = 5;
    private static final int CARS_COUNT = 30;
    private static final int TEACHER_CARS_COUNT = 7;
    private static final int BICYCLES_COUNT = 3;

    public static ArrayList<EntityInterface> generate() {
        ArrayList<EntityInterface> queue = new ArrayList<>();

        for (int i = 0; i <= ANONYMOUS_PEDESTRIANS_COUNT; i++) {
            queue.add(new Pedestrian());
        }

        for (int i = 0; i <= PEDESTRIANS_COUNT; i++) {
            queue.add(new Pedestrian(getRandomName()));
        }

        for (int i = 0; i <= CARS_COUNT; i++) {
            queue.add(new Car(getRandomPlateNumber()));
        }

        for (int i = 0; i <= TEACHER_CARS_COUNT; i++) {
            queue.add(new TeacherCar(getRandomPlateNumber()));
        }

        for (int i = 0; i <= BICYCLES_COUNT; i++) {
            queue.add(new Bicycle());
        }

        Collections.shuffle(queue);

        return queue;
    }

    private static String getRandomPlateNumber() {
        Random generator = new Random();
        return "DLX " + (generator.nextInt(89999) + 10000);
    }

    private static String getRandomName() {
        String[] names = {"John", "Jack", "James", "George", "Joe", "Jim"};
        return names[(int) (Math.random() * names.length)];
    }

}
